### PR TITLE
refactor/lagecy_logging

### LINKED ISSUE #1963: Clean up code related to legacy network logging
The old network reporter is disabled due to: 

https://github.com/equinor/ert/blob/efde2166e217b93fdbae568b6b0fc563ade4a49a/ert_logger/__init__.py#L18-L19

However, code for the old network reporter (https://github.com/equinor/ert/blob/main/job_runner/reporting/network.py) is still active, while it ultimately never logs anything. This should be clean up (either removed, or updated for a new logging solution).

### PR DESCRIPTION
**Pull request by sregales-TNO**
**Issue**
Resolves #1963

**Approach**
- Remove Network references
- Update tests and remove skipped ones
- Delete Network module

### CODE REVIEW

File: job_runner/reporting/misc.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from abc import ABC, abstractmethod
+
+
+class Report(ABC):
+    @abstractmethod
+    def report(self, msg):
+        raise NotImplementedError()
```
Comment (oyvindeide) [inline job_runner/reporting/misc.py:7]: Not really needed on `@abstractmethod`, could use `pass` or add a description in a doc string.

File: job_runner/cli.py [hunk 2]
```diff
@@ -1,13 +1,14 @@
 import argparse
+import json
 import os
 import signal
 import sys
 import json
 
 import job_runner.reporting as reporting
+from job_runner import JOBS_FILE
```
Comment (oyvindeide) [inline job_runner/cli.py:10]: Not related to the commit text

File: ert_logger/__init__.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
-import os, pwd
```
Comment (oyvindeide) [inline ert_logger/__init__.py:1]: Not related to the commit text

File: job_runner/reporting/file.py [hunk 4]
```diff
@@ -142,36 +135,35 @@ def _add_log_line(self, job):
...
     def _dump_error_file(self, job, error_msg):
-        fileH = open(self.ERROR_file, "a")
-        now = time.localtime()
-        fileH.write("<error>\n")
-        fileH.write(
-            "  <time>{:02d}:{:02d}:{:02d}</time>\n".format(
-                now.tm_hour, now.tm_min, now.tm_sec
+        with open(self.ERROR_file, "a") as fileH:
+            now = time.localtime()
+            fileH.write("<error>\n")
```
Comment (oyvindeide) [inline job_runner/reporting/file.py:140]: While you are refactoring might as well change variable name

**Review by oyvindeide — COMMENTED** [review]
Would be nice to have pure clean up in a separate commit, now there is some in most of them, sometimes touching different files from what it says in the commit text, which is a bit confusing. The whole `ert_logger` module could be cleaned as it relates to the legacy network logger.

File: job_runner/cli.py (hunk 2 above)
Comment (sregales-TNO) [reply job_runner/cli.py:10]: I think this was the doing of isort (which I know we don't use) sorting and cleaning unused imports

File: ert_logger/__init__.py (hunk 3 above)
Comment (sregales-TNO) [reply ert_logger/__init__.py:1]: I think this was the doing of isort (which I know we don't use)

File: job_runner/reporting/misc.py (hunk 1 above)
Comment (sregales-TNO) [reply job_runner/reporting/misc.py:7]: removed

File: job_runner/reporting/file.py (hunk 4 above)
Comment (sregales-TNO) [reply job_runner/reporting/file.py:140]: Why is it called "fileH"? is it something to do with "heading"?

Comment (oyvindeide) [reply job_runner/reporting/file.py:140]: No clue, suspect `file_handle`, but you are free to name it something more sensible.

File: job_runner/reporting/file.py [hunk 5]
```diff
@@ -142,36 +135,35 @@ def _add_log_line(self, job):
...
-                    if stderr:
-                        stderr_file = os.path.join(os.getcwd(), job.std_err)
-                    else:
-                        stderr = "<Not written by:{}>\n".format(job.name())
+            fileH.write("  <job>{}</job>\n".format(job.name()))
+            fileH.write("  <reason>{}</reason>\n".format(error_msg))
+            stderr_file = None
+            if job.std_err:
+                if os.path.exists(job.std_err):
+                    with open(job.std_err, "r") as errH:
```
[the remark this answers is not in the document]
Comment (oyvindeide) [reply job_runner/reporting/file.py:142]: Seems like a reasonable assumption 😊

File: job_runner/reporting/file.py [hunk 6]
```diff
@@ -116,70 +115,51 @@ def _complete_status_file(self, msg):
...
             # an arbitrary code less than -9.
-            if isinstance(msg, Start):
-                exit_code = -10
-            else:
-                exit_code = msg.exit_code
-
+            exit_code = -10 if isinstance(msg, Start) else msg.exit_code
             status = " EXIT: {}/{}".format(exit_code, msg.error_message)
         with open(self.STATUS_file, "a") as f:
             f.write("{:%H:%M:%S}  {}\n".format(msg.timestamp, status))
```
Comment (jondequinor) [inline job_runner/reporting/file.py:121]: `TIME_FORMAT`? Shouldn't you use that there?

File: job_runner/reporting/file.py [hunk 7]
```diff
@@ -116,70 +115,51 @@ def _complete_status_file(self, msg):
...
-            f.write(
-                "{:02d}:{:02d}:{:02d}  Calling: {} {}\n".format(
-                    now.tm_hour,
-                    now.tm_min,
-                    now.tm_sec,
-                    job.job_data["executable"],
-                    args,
-                )
-            )
+            time_str = time.strftime(TIME_FORMAT, time.localtime())
```
Comment (jondequinor) [inline job_runner/reporting/file.py:126]: Is this producing e.g. `03:01:02`?

File: job_runner/reporting/file.py (hunk 6 above)
Comment (jondequinor) [reply job_runner/reporting/file.py:121]: And maybe use an f-string?

File: job_runner/reporting/misc.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
+from abc import ABC, abstractmethod
+
+
+class Report(ABC):
+    @abstractmethod
+    def report(self, msg):
+        """Method wil report messages given, when executed by job runner."""
```
Comment (jondequinor) [inline job_runner/reporting/misc.py:7]:
```suggestion
        """Report a message."""
```

File: job_runner/reporting/misc.py [hunk 9]
```diff
@@ -0,0 +1,8 @@
...
+        pass
```
Comment (jondequinor) [inline job_runner/reporting/misc.py:8]:
```suggestion
class Reporter(ABC):
    @abstractmethod
    def report(self, msg: Message):
        """Report a message."""
        pass
```

File: job_runner/reporting/file.py [hunk 10]
```diff
@@ -83,10 +86,6 @@ def report(self, msg):
                     msg.timestamp
                 )
                 self._dump_ok_file()
-            else:
-                # this has already been handled by earlier event
-                pass
-
```
Comment (jondequinor) [inline job_runner/reporting/file.py:89]: You need to keep some form of this. It's explaining why the failure isn't handled.

File: job_runner/reporting/misc.py [hunk 11]
```diff
@@ -0,0 +1,8 @@
+from abc import ABC, abstractmethod
```
Comment (jondequinor) [inline job_runner/reporting/misc.py:1]: `misc`? Maybe `base` instead, or simply `reporter.py`

File: job_runner/reporting/misc.py [hunk 12]
```diff
@@ -0,0 +1,8 @@
...
+
+
+class Report(ABC):
```
Comment (jondequinor) [inline job_runner/reporting/misc.py:4]: `reporting` is a gerund, and we refer to these `File` `Network` and `Event` as reporters, so maybe just call it a `Reporter`?

File: job_runner/reporting/misc.py (hunk 11 above)
Comment (sregales-TNO) [reply job_runner/reporting/misc.py:1]: Base sounds greate!

**Review by jondequinor — APPROVED** [review]
Looks good, thanks!

**Pull request merged**